lib/jekyll/static_file.rb

```diff
@@ -80,7 +80,11 @@ def write(dest)
 
       FileUtils.mkdir_p(File.dirname(dest_path))
       FileUtils.rm(dest_path) if File.exist?(dest_path)
-      FileUtils.cp(path, dest_path)
+      if @site.safe
```

Comment:
I am happy to accept this if you add `Jekyll.env == "production"` here.

Production builds should produce an artifact (the compiled site) that is standalone and does not have any external dependencies on other files. A symlink cannot be guaranteed to validate that requirement.